--- a/src/head.c
+++ b/src/head.c
@@ -35,12 +35,14 @@ static int HeaderPrinted=0;
 static int PrintHeader=1;
 static CookieList *top = NULL;
 static char *cont_type = NULL;
+static char *cont_disp = NULL;
 
 void php_init_head(void) {
 	HeaderPrinted=0;
 	PrintHeader=1;
 	top = NULL;
 	cont_type = NULL;
+	cont_disp = NULL;
 }
 
 void NoHeader(void) {
@@ -131,7 +133,9 @@ void Header(void) {
 	if(r) {
 		*r='\0';
 		if(!strcasecmp(s->strval,"content-type")) {
-			cont_type = estrdup(0,r+1);;
+			cont_type = estrdup(0,r+1);
+		} else if (!strcasecmp(s->strval,"content-disposition")) {
+			cont_disp = estrdup(0,r+1);
 		} else {
 			*r=':';
 			ParseEscapes(s->strval);
@@ -272,6 +276,11 @@ void php_header(int type,char *str) {
 			} else {
 				fputs("Content-type:",stdout);
 				fputs(cont_type,stdout);
+				if (cont_disp) {
+					fputs("\015\012",stdout);
+					fputs("Content-Disposition: ",stdout);
+					fputs(cont_disp,stdout);
+				}
 				fputs("\015\012\015\012",stdout);
 			}
 		}
